tests/zstreamtest.c

```diff
@@ -2356,6 +2356,58 @@ static int basicUnitTests(U32 seed, double compressibility, int bigTests)
     }
     DISPLAYLEVEL(3, "OK \n");
 
+    DISPLAYLEVEL(3, "test%3i : Testing external sequence producer with static CCtx: ", testNb++);
+    {
+        size_t const dstBufSize = ZSTD_compressBound(CNBufferSize);
+        BYTE* const dstBuf = (BYTE*)malloc(dstBufSize);
+        size_t const checkBufSize = CNBufferSize;
+        BYTE* const checkBuf = (BYTE*)malloc(checkBufSize);
+        ZSTD_CCtx_params* params = ZSTD_createCCtxParams();
+        ZSTD_CCtx* staticCCtx;
+        void* cctxBuf;
+        EMF_testCase seqProdState;
+
+        CHECK_Z(ZSTD_CCtxParams_setParameter(params, ZSTD_c_validateSequences, 1));
+        CHECK_Z(ZSTD_CCtxParams_setParameter(params, ZSTD_c_enableSeqProducerFallback, 0));
+        ZSTD_CCtxParams_registerSequenceProducer(params, &seqProdState, zstreamSequenceProducer);
```

Comment:
While other parameter settings functions, such as `ZSTD_CCtxParams_setParameter()`, can fail,
`ZSTD_CCtxParams_registerSequenceProducer()` specifically cannot fail.
There is probably a topic around interface consistency here.

Reply:
Note that if we allow `ZSTD_CCtxParams_registerSequenceProducer()` to fail, then we also have to allow `ZSTD_registerSequenceProducer()` to fail.

Reply:
Yes, I guess they both should follow the same convention.